opsdroid/connector/matrix/connector.py

```diff
@@ -145,6 +165,11 @@ async def connect(self):
             )
             return
 
+        if not self.device_id:
```

Comment:
By the time we make it here if the user hasn't given a device id, we have one assigned to us. Does this get added to the store? Could we save it with the rest of the store and restore it next time opsdroid is started, this would at least reduce the amount of devices we create?

Reply:
I couldn't find a way to access the store through nio without first creating a client object and creating it assigns a new device_id